Code/Framework/AzCore/AzCore/std/smart_ptr/intrusive_ptr.h

```diff
@@ -277,11 +277,11 @@ namespace AZStd
         return const_cast<T*>(p.get());
     }
 
-    template<class T, class U>
-    intrusive_ptr<T> dynamic_pointer_cast(intrusive_ptr<U> const& p)
-    {
-        return azdynamic_cast<T*>(p.get());
-    }
+//    template<class T, class U>
+//    intrusive_ptr<T> dynamic_pointer_cast(intrusive_ptr<U> const& p)
+//    {
+//        return azdynamic_cast<T*>(p.get());
+//    }
```

Comment:
What is the reason for commenting this function out?

Reply:
This is the reason I asked about architecture layering on `#sig-core`. The code here is using azdynamic_cast which is a part of ?higher? level of abstraction + azdynamic_cast has specialization for instrusive_ptrs anyway?

Reply:
Well this function is for replicating an API that models the C++ standard library API for [shared_ptr](https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast)

So the function should remain uncommented.

Reply:
Ouch, I hoped we could've removed `#include <AzCore/RTTI/RTTI.h>` from this header. Oh well, I'll restore that.